### ISSUE TITLE
using request's ParameterBody in response

### ISSUE THREAD

**Issue by ravenblackdusk** [mock-server/mockserver#948]
how can i use the request's `ParameterBody` in response?
i've written a code like this and i'm pretty sure it used to work before i made some changes:
```
mockServer.`when`(
                request(path).method(POST)
                        .withHeader(HttpHeaders.CONTENT_TYPE, ContentType.APPLICATION_FORM_URLENCODED.mimeType)
                        .withBody(ParameterBody(
                                Parameter("client_id", clientId),
                                Parameter("client_secret", clientSecret),
                                Parameter("token"),
                        ))
        ).respond{
            jsonResponse().jsonBody(
                            (it.body as ParameterBody).value.getValues("token").first()
            )
        }
)
```
but `it.body` is now of type `StringBody` instead of `ParameterBody`.
is it possible to do something like this in MockServer?

**Comment by jamesdbloom** [comment]
This specific code has never been possible but you can achieve similar behaviour using a response template as documented here: https://mock-server.com/mock_server/response_templates.html or using method callbacks: https://mock-server.com/mock_server/creating_expectations.html#button_response_method_or_closure_callback or class callbacks: https://mock-server.com/mock_server/creating_expectations.html#button_response_class_callback

**Comment by ravenblackdusk** [comment]
@jamesdbloom thanks for your response. But in my code snippet I am using method callback(kotlin):
```
        .respond{
            jsonResponse().jsonBody(
                            (it.body as ParameterBody).value.getValues("token").first()
            )
        }
```
however `it` in the closure which is of type `HttpRequest`, `.getBody()` is of type `StringBody` instead of `ParameterBody`. Currently I'm using regex to get the parameter that I'm looking for:
```
"^.*&token=(.*)$".toRegex().find(it.bodyAsString)?.destructured?.toList()?.first()
```
I thought it would be much cleaner if the request body could be parsed into `ParameterBody` somehow:
```
(it.body as ParameterBody).value.getValues("token").first()
```
but I don't know how to do that.
In short I'm trying to get the request parameters of APPLICATION_FORM_URLENCODED in my response callback.
